### PR TITLE
fix: generate valid file name for virtual module chunk

### PR DESCRIPTION
**Pull request by underfin**

### CODE REVIEW

File: crates/rolldown_utils/src/path_ext.rs [hunk 1]
```diff
@@ -44,6 +44,11 @@ impl PathExt for std::path::Path {
       _ => file_name,
     };
 
+    // Avoid file name contained an unexpected NUL byte.
```
Comment (hyf0) [inline crates/rolldown_utils/src/path_ext.rs]: Should we do it in the begainning? I saw people return `\0[abslolute path]` in `resolve_id` hook.

Comment (underfin) [reply crates/rolldown_utils/src/path_ext.rs]: If normalized it at module loader resolve pharse, it maybe caused `debug_id` is incorrect and `RenderChunk#modules` is incorrect

Comment (hyf0) [reply crates/rolldown_utils/src/path_ext.rs]: I mean should we put this logic ahead of this function. https://github.com/rolldown/rolldown/pull/1782#discussion_r1694878474

File: crates/rolldown_utils/src/path_ext.rs [hunk 2]
```diff
@@ -59,4 +64,7 @@ fn test_representative_file_name() {
 
   let path = cwd.join("vue").join("mod.ts");
```
Comment (hyf0) [inline crates/rolldown_utils/src/path_ext.rs:65]: Consider test case 

```rust
let path = format!("\0{}", cwd.join("vue").join("mod.ts").to_string().unwrap().to_string()).
```

Comment (underfin) [reply crates/rolldown_utils/src/path_ext.rs:65]: Fixed it.

**Review by hyf0 — APPROVED** [review]

**Pull request merged**